store/src/errors.rs

```diff
@@ -41,7 +58,7 @@ pub enum StateInitializationError {
     #[error("Database error: {0}")]
     DatabaseError(#[from] DatabaseError),
     #[error("Failed to create nullifiers tree: {0}")]
-    FailedToCreateNullifiersTree(MerkleError),
+    FailedToCreateNullifiersTree(NullifierTreeError),
```

Comment:
minor nit: I'd use `FailedToCreateNullifierTree` (i.e., `Nullifier` without the `s`).